pkg/middleware/instrumentation.go

```diff
@@ -0,0 +1,109 @@
+package middleware
```

Comment:
NIT: Can we name this middleware file to more accurately reflect it is for prometheus/metrics? `instrumentation.go` is vague in a vacuum when in broader `middleware` package and not a more focused `prometheus` package where that term makes more sense in context.

Reply:
Can do.